webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/internal/RecentChangesAPI.scala

```diff
@@ -0,0 +1,157 @@
+/*
+*************************************************************************************
+* Copyright 2021 Normation SAS
+*************************************************************************************
+*
+* This file is part of Rudder.
+*
+* Rudder is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 3 of the License, or
+* (at your option) any later version.
+*
+* In accordance with the terms of section 7 (7. Additional Terms.) of
+* the GNU General Public License version 3, the copyright holders add
+* the following Additional permissions:
+* Notwithstanding to the terms of section 5 (5. Conveying Modified Source
+* Versions) and 6 (6. Conveying Non-Source Forms.) of the GNU General
+* Public License version 3, when you create a Related Module, this
+* Related Module is not considered as a part of the work and may be
+* distributed under the license agreement of your choice.
+* A "Related Module" means a set of sources files including their
+* documentation that, without modification of the Source Code, enables
+* supplementary functions or services in addition to those offered by
+* the Software.
+*
+* Rudder is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Rudder.  If not, see <http://www.gnu.org/licenses/>.
+
+*
+*************************************************************************************
+*/
+
+package com.normation.rudder.rest.internal
+
+import com.normation.errors.IOResult
+import com.normation.rudder.domain.policies.RuleId
+import com.normation.rudder.domain.policies.RuleUid
+import com.normation.rudder.rest.RestUtils._
+import com.normation.rudder.services.reports.NodeChangesService
+import com.normation.utils.DateFormaterService
+import net.liftweb.common.Loggable
+import net.liftweb.common._
+import net.liftweb.http.rest.RestHelper
+import net.liftweb.json.JsonAST.JValue
+import net.liftweb.json.JsonDSL._
+import org.joda.time.DateTime
+import org.joda.time.Interval
+import com.normation.errors._
+import com.normation.rudder.domain.reports.ResultRepairedReport
+import com.normation.box._
+import com.normation.inventory.domain.NodeId
+import com.normation.rudder.domain.policies.DirectiveId
+import com.normation.rudder.domain.policies.DirectiveUid
+import zio.syntax.ToZio
+
+import scala.util.Random
+
+class RecentChangesAPI (
+   nodeChangesService: NodeChangesService
+) extends RestHelper with Loggable {
```

Comment:
Can you use the endpoint definition that we usually use, not the liftweb specific one? (I understand it was for a quick test, but it needs to be changed for the final PR)

Reply:
We don't use our endrpoint definition for internal and it can't be integrated this way

Reply:
yes we use it for internal API, see for example: `NodeDisplayInheritedProperties` in `NodeApi`. Perhaps we didn't ported all existing internal API the new (sort of) schema, but it should be done, at least for new ones. 

I don't see why we couldn't integrate that API this way, what is the technical blocking point ?